src/librustc_data_structures/dfx.rs

```diff
@@ -0,0 +1,54 @@
+pub use rustc_hash::FxHashMap;
+use std::borrow::Borrow;
+use std::hash::Hash;
+
+
+/// A wrapper around FxHashMap that does not provide iteration support.
+///
+/// It supports insert, remove, get and get_mut functions from FxHashMap.
+pub struct DFxHashMap<K, V>
+{
+    base: FxHashMap<K, V>,
+}
+
+impl<K: Hash + Eq, V> DFxHashMap<K, V>
+{
+    pub fn new() -> DFxHashMap<K, V>
+    {
+        DFxHashMap{base: FxHashMap::default()}
+    }
+}
+
+impl<K, V> DFxHashMap<K, V>
+    where
+    K: Eq + Hash,
+{
+    pub fn get<Q: ?Sized>(&self, k: &Q) -> Option<&V>
+    where
+        K: Borrow<Q>,
+        Q: Hash + Eq,
+    {
+        self.base.get(k)
+    }
+
+    pub fn get_mut<Q: ?Sized>(&mut self, k: &Q) -> Option<&mut V>
+    where
+        K: Borrow<Q>,
+        Q: Hash + Eq,
+    {
+        self.base.get_mut(k)
+    }
+
+    pub fn insert(&mut self, k: K, v: V) -> Option<V>
+    {
+        self.base.insert(k, v)
+    }
+
+    pub fn remove<Q: ?Sized>(&mut self, k: &Q) -> Option<V>
+    where
+        K: Borrow<Q>,
+        Q: Hash + Eq,
+    {
+        self.base.remove(k)
+    }
```

Comment:
Please add support for the entry function from `FxHashMap`.

Reply:
OK.